### PR TITLE
✨Add redirect logic for public paths

### PR DESCRIPTION
**Pull request by anncatton**
**Description of changes**

Adds redirect logic to Navbar login button
- including for private and public paths, and no path (this should redirect to `logged-in`)
- removes `/` default value for GoogleLogin `redirectPath`, this caused a 403 on oauth because it add a trailing `/` when the original path was `/login`
- corrects a typo on the Create Program form

**Type of Change**

- [x] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [x] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: components/NavBar.tsx [hunk 1]
```diff
@@ -60,7 +63,32 @@ export default function Navbar({ hideLink }: { hideLink?: boolean }) {
   const { EGO_URL } = getConfig();
   const { token: egoJwt, logOut, data: userModel } = useAuthContext();
   const canAccessSubmission = !!egoJwt && (canReadSomeProgram(egoJwt) || isRdpcMember(egoJwt));
-  const { asPath: path } = usePageContext();
+  const { asPath: path, query } = usePageContext();
+
+  const [loginPath, setLoginPath] = React.useState('');
```
Comment (hlminh2000) [inline components/NavBar.tsx]: I don't think this has to be a state? can just be `const loginPath = ...`. For optimization, `React.useMemo` can be used.

File: components/NavBar.tsx [hunk 2]
```diff
@@ -60,7 +63,32 @@ export default function Navbar({ hideLink }: { hideLink?: boolean }) {
...
+    if (redirect) {
+      const parsedRedirect = queryString.parseUrl(redirect);
+      const existingQuery = queryString.stringify(parsedRedirect.query);
+
+      const queryRedirect = createRedirectURL({
+        origin: location.origin,
+        path: parsedRedirect.url,
+        query: existingQuery,
+      });
+      setLoginPath(`${EGO_URL}${queryRedirect}`);
```
Comment (hlminh2000) [inline components/NavBar.tsx]: let's use `url-join` for safety :)

File: components/NavBar.tsx [hunk 3]
```diff
@@ -60,7 +63,32 @@ export default function Navbar({ hideLink }: { hideLink?: boolean }) {
...
+    } else if (path === '/' || path === '/login') {
+      setLoginPath(EGO_URL);
+    } else {
+      const redirect = createRedirectURL({
+        origin: location.origin,
+        path,
+      });
+      setLoginPath(`${EGO_URL}${redirect}`);
```
Comment (hlminh2000) [inline components/NavBar.tsx]: `url-join`

File: components/NavBar.tsx (hunk 1 above)
Comment (anncatton) [reply components/NavBar.tsx]: If I remove the effect I run into the issue with the location object not being defined yet...

**Pull request merged**